tests/integration-tests/conftest.py

```diff
@@ -1182,18 +1182,19 @@ def odcr_stack(request, region, placement_group_stack, cfn_stacks_factory, vpc_s
     public_subnet = vpc_stack.get_public_subnet()
     public_subnets = vpc_stack.get_all_public_subnets().copy()
     public_subnets.remove(public_subnet)
-    availability_zone = boto3.resource("ec2").Subnet(public_subnet).availability_zone
-    availability_zone_2 = boto3.resource("ec2").Subnet(public_subnets[0]).availability_zone
+    default_or_random_public_az = boto3.resource("ec2").Subnet(public_subnet).availability_zone
+    availability_zone_1 = boto3.resource("ec2").Subnet(public_subnets[0]).availability_zone
+    availability_zone_2 = boto3.resource("ec2").Subnet(public_subnets[1]).availability_zone
```

Comment:
[Nitpick] This whole method and how these resources are created and used should likely be redesigned. But this is out-of-scope for this issue.